lib/src/presentation/views/upcoming_movies_grid.dart

```diff
@@ -0,0 +1,104 @@
+import '../../domain/entity/movie_event.dart';
+import '../bloc/i_upcoming_bloc.dart';
+import '../bloc/upcoming_bloc.dart';
+import '../widgets/app_drawer.dart';
+import '../widgets/custom_progress_indicator.dart';
+import '../widgets/movie_grid_item.dart';
+import '../widgets/movie_list_header.dart';
+import '../widgets/plain_text.dart';
+
+class UpcomingMoviesGrid extends StatefulWidget {
+  UpcomingMoviesGrid({
+    super.key,
+    required this.upcomingBloc,
+  });
+
+  final IUpcomingBloc upcomingBloc;
+  static const String routeName = '/upcomingMoviesGrid';
+  static const String title = 'Upcoming movies';
+  static const String emptyMsg = 'Sorry, it seems that we don´t have any movies in this category to show you!';
+
+  @override
+  State<UpcomingMoviesGrid> createState() => UpcomingMoviesGridState();
+}
+
+class UpcomingMoviesGridState extends State<UpcomingMoviesGrid> {
+  @override
+  void initState() {
+    super.initState();
+    widget.upcomingBloc.getUpcoming();
+  }
+
+  @override
+  void dispose() {
+    widget.upcomingBloc.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        appBar: AppBar(
+          backgroundColor: UIConstants.backgroundColor,
+        ),
+        drawer: const AppDrawer(),
+        backgroundColor: UIConstants.backgroundColor,
+        body: StreamBuilder<MovieEvent>(
+          initialData: MovieEvent(
+            status: Status.loading,
+          ),
+          stream: widget.upcomingBloc.moviesStream,
+          builder: (BuildContext context, AsyncSnapshot<MovieEvent> snapshot) {
+            Status status = snapshot.data!.status;
+            switch (status) {
+              case Status.loading:
+                return const CustomProgressIndicator();
+              case Status.success:
+                return Column(
+                  children: [
+                    const MovieListHeader(
+                      title: UpcomingMoviesGrid.title,
+                    ),
+                    UIConstants.sectionSpace,
+                    Expanded(
+                      child: GridView.count(
+                        crossAxisCount: UIConstants.gridRowCount,
+                        crossAxisSpacing: UIConstants.gridHorizontalSpace,
+                        mainAxisSpacing: UIConstants.gridVerticalSpace,
+                        childAspectRatio: UIConstants.gridAspectRatio,
+                        children: List.generate(
+                            snapshot.data?.moviesList?.length ?? 0, (index) {
+                          return MovieGridItem(
+                            moviesList: snapshot.data!.moviesList!,
+                            index: index,
+                            gridRouteName: UpcomingMoviesGrid.routeName,
+                          );
+                        }),
+                      ),
+                    ),
+                  ],
+                );
```

Comment:
i saw this girdview above, it can be abstracted too

Reply:
done
(I left only one generic grid screen, that's used to display popular and upcoming movies)